app/logger_config.py

```diff
@@ -12,15 +14,60 @@ def format(self, record):
         else:
             record.url = None
             record.remote_addr = None
-
         return super().format(record)
 
 
+class CloudWatchHandler(logging.Handler):
+    def __init__(self, log_group, stream_name):
+        super().__init__()
+        self.client = boto3.client("logs")
+        self.log_group = log_group
+        self.stream_name = stream_name
+
+    def emit(self, record):
+        try:
+            log_entry = self.format(record)
+            self.client.put_log_events(
+                logGroupName=self.log_group,
+                logStreamName=self.stream_name,
+                logEvents=[
+                    {
+                        "timestamp": int(record.created * 1000),
+                        "message": log_entry,
+                    }
+                ],
+            )
+        except ClientError as e:
+            print(f"Error sending log to CloudWatch: {e}")
+
+
 def setup_logging(app):
     formatter = RequestFormatter(
         "[%(asctime)s] %(remote_addr)s requested %(url)s\n"
         "%(levelname)s in %(module)s: %(message)s"
     )
-    default_handler.setFormatter(formatter)
 
-    app.logger.setLevel(logging.INFO)
+    app_logger = logging.getLogger("app_logger")
+    app_logger.setLevel(logging.INFO)
+    audit_logger = logging.getLogger("audit_logger")
+    audit_logger.setLevel(logging.INFO)
+
+    if os.getenv("ENV") == "aws":
```

Comment:
I think we can hook into the existing CONFIG_SOURCE env var which we could use to distinguish between AWS deployment and non.

Ideally we could keep all AWS specific code in the Config classes but I don't think we can define `app_handler` and `audit_handler` on it, as the flask config object needs to be serialisable so the properties can't be objects.

aside: our calls to s3 for getting objects are directly in the routes so its not actually true that all boto calls are abstracted to the Config classes but as much as possible would be ideal. (I'd like to abstract the s3 calls to a FileHandler class; we could then follow a similar pattern to the Config class there and in any other place where the call to AWS can't be made at startup)

Reply:
unneeded now 👌

Reply:
Discussed PR on call